flycontrols.js

```diff
@@ -0,0 +1,183 @@
+const Signal = require('signals')
+const mat4 = require('pex-math/mat4')
+const vec3 = require('pex-math/vec3')
+const quat = require('pex-math/quat')
+const euler = require('pex-math/euler')
+
+
+function Flycontrols (opts) {
+  this.type = 'Flycontrols'
+  this.enabled = true
+  this.changed = new Signal()
+  this.entity = null
+  this.dirty = false
+  this.camera =  false
+  this.currentYaw = [0,0,0,1]
+  this.currentPitch = [0,0,0,1]
+
+  this.isLocked = false
+
+  this.moveForward = false 
+  this.moveLeft = false
+  this.moveBackward = false
+  this.moveRight = false
+  this.moveUp = false
+  this.moveDown = false
+
+  const initialState = {
```

Comment:
All these properties don't require an extra object: `this.posision` etc is sufficient. (we should do that in the orbiter too).